main/inc/lib/sessionmanager.lib.php

```diff
@@ -5430,4 +5430,66 @@ public static function getCoursesListByCourseCoach($coachId)
         ));
     }
 
+    /**
+     * Get a session list filtered by name, description or short description extra field
+     * @param string $term The term to search
+     * @return array The list
+     */
+    public static function searchSession($term)
+    {
+        $sTable = Database::get_main_table(TABLE_MAIN_SESSION);
+        $sfvTable = Database::get_main_table(TABLE_MAIN_SESSION_FIELD_VALUES);
+
+        $term = Database::escape_string($term);
+
+        $resultData = Database::select('id, name, date_start, date_end, duration, description', $sTable, array(
+            'where' => array(
+                "name LIKE %?% " => $term,
+                "OR description LIKE %?% " => $term,
+                "OR id IN (
+                    SELECT session_id
+                    FROM $sfvTable
+                    WHERE field_value LIKE %?%
+                ) " => $term
+            )
+        ));
+
+        $fieldsArray = array(
+            'short_description', 'mode', 'human_text_duration', 'vacancies', 'brochure', 'target', 'schedule'
```

Comment:
You cannot set fixed extra fields.
I would suggest you add one parameter to the function ($alsoSearchExtraFields, false by default) to decide whether you should search in extra fields or not *and* that you search *all* session extra fields.